src/utils/DisabledDate.js

```diff
@@ -0,0 +1,243 @@
+
+          return (
+            hasFrom && hasTo && date < thisRange.to && date > thisRange.from
+          )
+        })
+      },
+      customPredictor: () => {
+        return has.customPredictor && dd.customPredictor(date)
+      },
+      specificDate: () => {
+        if (!has.specificDates) return false
+
+        return dd.dates.some((d) => {
+          return this._utils.compareDates(date, d)
+        })
+      },
+      daysOfWeek: () => {
+        if (!has.daysOfWeek) return false
+
+        return dd.days.indexOf(this._utils.getDay(date)) !== -1
+      },
+      daysOfMonth: () => {
+        if (!has.daysOfMonth) return false
+
+        return dd.daysOfMonth.indexOf(this._utils.getDate(date)) !== -1
+      },
+    }
+  }
+
+  isMonthDisabledVia(date) {
+    const { config } = this
+    const { has } = config
+    const { to } = config
+    const { from } = config
+    const month = this._utils.getMonth(date)
+    const year = this._utils.getFullYear(date)
+
+    return {
+      to: () => {
+        const isYearInPast = has.to && year < to.year
+
+        if (isYearInPast) {
+          return true
+        }
+
+        return has.to && month < to.month && year <= to.year
+      },
+      from: () => {
+        const isYearInFuture = has.from && year > from.year
+
+        if (isYearInFuture) {
+          return true
+        }
+
+        return has.from && month > from.month && year >= from.year
+      },
+    }
+  }
+
+  isYearDisabledVia(date) {
+    const { has } = this.config
+    const { to } = this.config
+    const { from } = this.config
+    const year = this._utils.getFullYear(date)
+
+    return {
+      to: () => {
+        return has.to && year < to.year
+      },
+      from: () => {
+        return has.from && year < from.year
+      },
+    }
+  }
+
+  /**
+   * Checks if the given date should be disabled
+   * @param {Date} date
+   * @return {Boolean}
+   */
+  // eslint-disable-next-line complexity,max-statements
+  isDateDisabled(date) {
+    if (!this.config.exists) return false
+
+    const isDisabledVia = this.isDateDisabledVia(date)
+
+    if (isDisabledVia.to()) {
+      return true
+    }
+
+    if (isDisabledVia.from()) {
+      return true
+    }
+
+    if (isDisabledVia.range()) {
+      return true
+    }
+
+    if (isDisabledVia.specificDate()) {
+      return true
+    }
+
+    if (isDisabledVia.daysOfWeek()) {
+      return true
+    }
+
+    if (isDisabledVia.daysOfMonth()) {
+      return true
+    }
+
+    return isDisabledVia.customPredictor()
+  }
+
+  /**
+   * Checks if the given month should be disabled
+   * @param {Date} date
+   * @return {Boolean}
+   */
+  // eslint-disable-next-line complexity,max-statements
+  isMonthDisabled(date) {
+    const { config } = this
+    const isDisabledVia = this.isMonthDisabledVia(date)
+
+    if (!config.exists) {
+      return false
+    }
+
+    if (isDisabledVia.to()) {
+      return true
+    }
+
+    if (isDisabledVia.from()) {
```

Comment:
What about: 
```js
if (
      isDisabledVia.to() ||
      isDisabledVia.from()
    ) {
      return true
    }
```

Reply:
Yes, `prettier` prefers it on one line:

    if (isDisabledVia.to() || isDisabledVia.from()) {
      return true
    }